Add buffer size API and UI for trends

cpascual opened the pull request:
- Add `bufferSize()` and `setBufferSize()` to `TaurusTrendSet`
- Add `BufferSizeTool` (and use it in `TaurusTrend` by default`)
- Add `-b` /  `--buffer` option to `tpg trend` CLI

Commit:
Add setBufferSize API to TaurusTrendSet

Add setBufferSize() and bufferSize() to TaurusTrendSet

Commit:
Add MaxBufferTool

Add a Tool to provide a GUI control of the buffer size
of trends

Commit:
Add the buffer tool to TaurusTrend

Commit:
Update check-list in README.md

Mark the "UI for curve buffers" item as done

Commit:
remname "MaxBufferTool" -> "BufferSizeTool"

Rename and class and file name

Commit:
Add TaurusTrendSet.clearBuffer()

Add method to remove existing points. Use it
when downsizing the max buffer.

Commit:
Add buffer_size arg to TaurusTrend constructor

Allow to pass the buffer_size as an argument of
the TaurusTrend widget

Commit:
Add buffer CLI option to taurus tpg trend command

Allow to set the buffer size via the -b / --buffer
CLI option

Commit:
m, flake8

Commit:
m, Fix tooltip

guifrecuni commented:
Thanks Carlos the spinbox option looks good.
There seems to be some issues while changing sizes (which may go to downsizing while editing)

cpascual commented:
For the problem of accidentally losing info due to buffer clearing when downsizing, here are possible mitigators:
- show a warning dialog requesting confirmation before downsizing
- do not use valueChanged but editingFinished to trigger a change
- do not completely clear buffers, keep the last n data (being n the new size)

Also, we found a serious issue: buffer clearing resets curve properties (color, symbols, etc are lost)

Commit:
Update into buffersizetool branch with latest changes in master

Fix conflicts for:
	README.md
	taurus_pyqtgraph/cli.py
	taurus_pyqtgraph/taurustrendset.py
	taurus_pyqtgraph/trend.py

Commit:
WIP: 1st implementation of setMaxDataBufferSize

Do a naive implementation of setMaxDataBufferSize.
WIP: it does not work as expected with -b because it
only affects to trend sets already present at the time of calling

Commit:
Make buffer tool accessible from the TaurusTrend object

This is useful in order to programatically change the buffer size
of a trend widget

Commit:
Enable the buffer limit also on newly added trendsets

Allow the buffer tool to react on newly added trend sets and apply
the limit to them.

This requires https://github.com/pyqtgraph/pyqtgraph/pull/1388 to be
integrated in pyqtgraph.

Commit:
Allow downsizing of trendset buffers without clearing

If setBufferSize is called with a size smaller than the current buffer
size, the buffer is cleared, destroying all data in it, which is not
desirable. Instead, keep the maximum possible newest data that
accomodates in the new size.

Commit:
Trigger buffer tool on editing finished

The buffer tool action is triggered on value changed, causing
problems with values being applied while editing (potentially
destroying data). Instead, trigger on editing finished.

Commit:
Use a menu for buffer action instead of a spinbox

Change the buffer tool from a widget action with an spinbox to a
QAction that triggers a dialog. This avoids some weird UI glitches
in which the menu disappears when clicking the spin box for edit.

cpascual commented:
Regarding [this problem](https://github.com/taurus-org/taurus_pyqtgraph/pull/47#issuecomment-626852461), I addressed it by allowing downsizing without total wipe and changing from a spinbox to a dialog for entering the new value
